eng/scripts/Test-Code.ps1

```diff
@@ -73,67 +97,79 @@ function CopyNativeBinaryToTestDirs {
     param(
         [Parameter(Mandatory=$true)]
         [string]$nativeBinaryPath,
-        [string[]]$areas = @()
+        [string[]]$testsRootDirs
     )
     Write-Host "Copying native AzureMcp to test directories"
 
-    if (!$areas) {
-        $testDirectories = Get-ChildItem -Path $RepoRoot -Recurse -Filter "*.LiveTests" -Directory
-    } else {
-        $testDirectories = @()
-        foreach ($area in $areas) {
-            $areaName = $area.ToLower()
-            $areaPath = $areaName -eq 'core' ? "$RepoRoot/core/tests" : "$RepoRoot/areas/$areaName/tests"
-            if (Test-Path $areaPath) {
-                $areaTestDirectories = Get-ChildItem -Path $areaPath -Recurse -Filter "*.LiveTests" -Directory
-                $testDirectories += $areaTestDirectories
-            }
-        }
+    $testProjectDirs = @()
+    foreach ($testsRootDir in $testsRootDirs) {
+        $areaTestProjectDirs = Get-ChildItem -Path $testsRootDir -Recurse -Filter "*.LiveTests" -Directory
+        $testProjectDirs += $areaTestProjectDirs
     }
 
-    foreach ($testDir in $testDirectories) {
+    foreach ($testDir in $testProjectDirs) {
         $targetDirectory = "$($testDir.FullName)/bin/Debug/net9.0"
         Copy-Item $nativeBinaryPath $targetDirectory -Force
     }
 }
 
-function AddTestProjects($path) {
-    if($TestType -in @('Live', 'All')) {
-        $script:testProjects += Get-ChildItem $path -Recurse -File -Filter "*.LiveTests.csproj"
+function CreateTestSolution {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$workPath,
+        [Parameter(Mandatory=$true)]
+        [string[]]$testsRootDirs,
+        [Parameter(Mandatory=$true)]
+        [string]$testType
+    )
+
+    $testProjects = @()
+    foreach ($testsRootDir in $testsRootDirs) {
+        if($testType -in @('Live', 'All')) {
```

Comment:
Any reason to inspect `$testType` within the loop when it won't change during the scripts evaluation? Couldn't we turn this into a local variable for the lifetime of the method call?